src/core/qgsvectorlayer.cpp

```diff
@@ -2387,6 +2387,19 @@ long QgsVectorLayer::featureCount() const
          ( mEditBuffer ? mEditBuffer->mAddedFeatures.size() - mEditBuffer->mDeletedFeatureIds.size() : 0 );
 }
 
+long QgsVectorLayer::featureCount( QgsFeatureRequest request )
+{
+  QgsFeatureIterator fit = getFeatures( request );
```

Comment:
Hi @rldhont, Yes, I see, but you could create a new QgsFeatureRequest using this as seed, but removing all no-necessary attributes to fetch, You only want count features. Sorry, I have explained it better?

QgsFeatureRequest r = new QgsFeatureRequest(....);
r.setFlags( QgsFeatureRequest::NoGeometry );
r.setSubsetOfAttributes( QgsAttributeList() ) );